### DISCUSSION TITLE
[Projects Beta] [Enhancement] Allow group by label

### DISCUSSION THREAD

**Discussion by danielinux7** [community/community#9172] [Projects and Issues]
Hello,

Is it possible to add the ability to group by labels in a view

Best Regards,
Nart.

**Comment by altela** [comment]
Hi @danielinux7 
I'm pretty sure this is what you asked.

![Screen Shot 2021-12-22 at 14 40 06](https://user-images.githubusercontent.com/68892527/147047354-e9da5c96-13b7-4c3b-b37e-bde30fc74a46.png)

First you need to click the magnifying glass on the left, then type `label:enhancement`. You can also change it to any label that you've added on the repo. If you want to combine it with `issue` status, add either `is:close` or `is:open`

Anyway here's other example of mine : 

![Screen Shot 2021-12-22 at 14 45 32](https://user-images.githubusercontent.com/68892527/147047890-32053543-6e07-4beb-b2f4-ea5282ef95d5.png)

**Comment by danielinux7** [comment]
Hello @Altela 
I meant something else, when you group by label from the drop down menu.
![IMG_20211223_175428](https://user-images.githubusercontent.com/2645266/147260161-c0cdfa02-20c1-4f1b-9549-d3ea1e701143.jpg)
You get a grouped by list of labels.
![IMG_20211223_181840](https://user-images.githubusercontent.com/2645266/147260249-31e76ee5-ec66-41ed-abe8-f09bd0085a06.jpg)
Best Regards,
Nart.

**Reply by jsmorabito** [reply]
+1 for this!

**Reply by iamhenry** [reply]
yes, would def make use of this a lot!

**Reply by archon810** [reply]
Surprised this is not a feature already. You can group by other fields, but not by label? I understand there could be a lot of permutations, but it's really no different than, say, grouping by assignee as there could be multiple assignees assigned as well, and this functionality is already implemented. So why not labels?

**Reply by zoebosschaart** [reply]
This would be cool!!

**Reply by viktorix** [reply]
+1

**Reply by marchershey** [reply]
Still yet a feature...

**Reply by essential-randomness** [reply]
Came here to search for this :(

**Reply by horiaionut** [reply]
+1

**Reply by cnatis** [reply]
+1

**Reply by katarinasupe** [reply]
+1

**Reply by okazymyrov** [reply]
+1

**Reply by kwajiehao** [reply]
+1

**Reply by khismatullin** [reply]
+1

**Reply by kwajiehao** [reply]
> Surprised this is not a feature already. You can group by other fields, but not by label? I understand there could be a lot of permutations, but it's really no different than, say, grouping by assignee as there could be multiple assignees assigned as well, and this functionality is already implemented. So why not labels?

Actually I think this might be because labels belong to a repo, whereas the other fields belong to the project board. To group by labels, you would need to pull metadata for every issue on the project board and merge labels from different repos, which is a lot of network calls (though I guess they do have a graphql api so it might just be one).

**Reply by MrCeliaki** [reply]
+1

**Reply by jab-ioki** [reply]
+1

**Comment by MrChocolateMoose** [comment]
This would be super helpful. Labels are superior to custom fields in github for organizing at this point so it would be good to group by.

**Comment by 0xBasically** [comment]
Yes please add this

**Comment by rileybroughten** [comment]
You can now slice by the `Label` field, so that you can drill into each label and view the associated items to it in your view - check out the changelog announcement [here](https://github.blog/changelog/2023-08-10-github-issues-projects-august-10th-update/#slice-by)!

<img width="1438" alt="Screenshot 2023-09-07 at 12 21 23 PM" src="https://github.com/community/community/assets/101840513/b77d7894-5e30-45a7-97f8-f9d7a402ebf6">

Would be curious to hear how this helps your use case!

**Reply by davidproxima** [reply]
how do you add this view of the labels that is in the left column?

**Reply by rileybroughten** [reply]
You can slice by a field in your project views - check out the documentation [here](https://docs.github.com/en/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/customizing-the-board-layout#slicing-by-field-values)!

**Reply by danielgomezrico** [reply]
Slicing helps but it does not allow you to see all grouped by the label which is still needed to me

**Comment by danemacmillan** [comment]
Any movement on this? Being able to treat repository issue labels like project fields would open up a lot of flexibility and decouple a lot of this project-specific management, which would be very conducive to automating aspects of projects if the data is all available from an issue.

**Comment by nickazg** [comment]
Yeah, slicing is ok but requires you to click. Why cant we get grouping in the same way you can group by status etc?
